azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/VirtualMachineToNodeMetadata.java

```diff
@@ -179,8 +180,15 @@ public String apply(Status input) {
       Credentials credentials = credentialStore.get("node#" + virtualMachine.name());
       builder.credentials(LoginCredentials.fromCredentials(credentials));
 
-      builder.publicAddresses(getPublicIpAddresses(virtualMachine.properties().networkProfile().networkInterfaces()));
-      builder.privateAddresses(getPrivateIpAddresses(virtualMachine.properties().networkProfile().networkInterfaces()));
+      Iterable<String> publicIpAddresses = getPublicIpAddresses(virtualMachine.properties().networkProfile().networkInterfaces());
+      Iterable<String> privateIpAddresses = getPrivateIpAddresses(virtualMachine.properties().networkProfile().networkInterfaces());
+
+      if (!Iterables.all(publicIpAddresses, Predicates.<String>isNull())) {
```

Comment:
Better fix the implementation of `getPublicIpAddresses` and `getPrivateIpAddresses` to skip over null values. This will reject the whole collection and there could be valid values in it.